stl/inc/xutility

```diff
@@ -2160,9 +2163,8 @@ namespace ranges {
         };
 
         template <class _Ty>
-        concept _Has_ADL = requires(_Ty&& __t) {
-            { _Fake_decay_copy(end(static_cast<_Ty&&>(__t))) } ->
-                sentinel_for<decltype(_RANGES begin(static_cast<_Ty&&>(__t)))>;
+        concept _Has_ADL = requires(_Ty& __t) {
+            { _Fake_decay_copy(end(__t)) } -> sentinel_for<decltype(_RANGES begin(__t))>;

```

Comment:
Why is `begin` qualified with `_RANGES` but `end` is not?

Reply:
Because they refer to different things: we want to determine if ADL for non-member `end` works and returns something that is a sentinel for "whatever `ranges::begin` returns". Qualifying both or qualifying neither would have a meaning other than what [range.access.end]1.3 wants.

Reply:
I see, thanks. (I wonder if we should have a comment whenever ADL is desired, although I wouldn't go as far as requesting an `_ADL` macro expanding to nothing.)

Reply:
> I wonder if we should have a comment whenever ADL is desired

This is what I had in mind when I suggested defining `_UNQ` ("unqualified") as an empty macro and using it to "qualify" the names of CPOs in the discussion that produced `_RANGES` a couple of weeks ago. `_UNQ end` here would very clearly be an explicit request for ADL.